src/loader.rs

```diff
@@ -25,28 +53,196 @@ impl AssetLoader for ObjLoader {
     }
 }
 
+#[cfg(feature = "scene")]
+impl FromWorld for ObjLoader {
+    fn from_world(world: &mut World) -> Self {
+        let supported_compressed_formats = match world.get_resource::<RenderDevice>() {
+            Some(render_device) => CompressedImageFormats::from_features(render_device.features()),
+
+            None => CompressedImageFormats::all(),
+        };
+        Self {
+            supported_compressed_formats,
+        }
+    }
+}
+
 #[derive(Error, Debug)]
 pub enum ObjError {
     #[error("Invalid OBJ file: {0}")]
-    InvalidFile(#[from] tobj::LoadError),
+    TobjError(#[from] tobj::LoadError),
+    #[cfg(feature = "scene")]
+    #[error("Invalid image file for texture: {0}")]
+    InvalidImageFile(PathBuf),
+    #[error("Asset reading failed: {0}")]
+    AssetIOError(#[from] bevy_asset::AssetIoError),
+    #[error("Texture conversion failed: {0}")]
+    TextureError(#[from] bevy_render::texture::TextureError),
 }
 
 async fn load_obj<'a, 'b>(
     bytes: &'a [u8],
     load_context: &'a mut LoadContext<'b>,
+    #[cfg(feature = "scene")] supported_compressed_formats: CompressedImageFormats,
 ) -> Result<(), ObjError> {
-    let mesh = load_obj_from_bytes(bytes)?;
-    load_context.set_default_asset(LoadedAsset::new(mesh));
+    #[cfg(not(feature = "scene"))]
+    let obj = load_obj_from_bytes(bytes)?;
+    #[cfg(feature = "scene")]
+    let obj = load_obj_from_bytes(bytes, load_context, supported_compressed_formats).await?;
+
+    load_context.set_default_asset(LoadedAsset::new(obj));
     Ok(())
 }
 
-fn load_mtl(_path: &std::path::Path) -> tobj::MTLLoadResult {
-    Err(tobj::LoadError::OpenFileFailed)
+#[cfg(feature = "scene")]
+async fn load_texture_image<'a, 'b>(
+    image_path: &'a str,
+    load_context: &'a mut LoadContext<'b>,
+    supported_compressed_formats: CompressedImageFormats,
+) -> Result<(Image, String), ObjError> {
+    let path = Path::new(&image_path);
```

Comment:
Use `set_file_name` to overwrite path, instead of using just the raw file name `&image_path`. This allows the loader to work with `http`/`https` resources, or with resources on paths deeper inside the asset directory.

Here's a working example to replace this line with:
```
    let asset_path = load_context.path();
    let img_path: &Path = asset_path.as_ref();
    let mut img_path = img_path.to_owned();
    img_path.set_file_name(image_path);
    let path = &img_path;
    log::info!("Loading Texture Image  path = {:?}", path);
```

Reply:
I am doing most of my work in a 0.9 branch and implemented something similar https://github.com/AmionSky/bevy_obj/pull/12/commits/e779c55ebea069e9a794c4839870bcb272e70c7e but I like your solution more since it doesn't add `unwrap` so did it here https://github.com/AmionSky/bevy_obj/pull/12/commits/d031b9a7f8256bdc0f9b1aa460845f9e37916158